crates/rpc/rpc-types/src/eth/transaction/mod.rs

```diff
@@ -65,3 +67,88 @@ pub struct Transaction {
     #[serde(rename = "type", skip_serializing_if = "Option::is_none")]
     pub transaction_type: Option<U64>,
 }
+
+impl Transaction {
+    /// Create a new rpc transction result, using the given block hash, number, and tx index fields
+    /// to populate the corresponing fields in the rpc result.
+    pub(crate) fn from_recovered_with_block_context(
+        tx: TransactionSignedEcRecovered,
+        block_hash: H256,
+        block_number: BlockNumber,
+        tx_index: U256,
+    ) -> Self {
+        let mut tx = Self::from_recovered(tx);
+        tx.block_hash = Some(block_hash);
+        tx.block_number = Some(U256::from(block_number));
+        tx.transaction_index = Some(tx_index);
+        tx
+    }
+
+    /// Create a new rpc transaction result from a signed and recovered transaction, setting
+    /// environment related fields to `None`.
+    ///
+    /// Sets the sender public key to `None` as well.
+    pub(crate) fn from_recovered(tx: TransactionSignedEcRecovered) -> Self {
```

Comment:
this is essentially a `Pending` transaction, right?
should include that in function name or docs

Reply:
yeah, this would be a pending transaction, will add to docs